File: paper/src/main/java/com/al3x/housing2/Data/StatActionData.java
package com.al3x.housing2.Data;

import com.al3x.housing2.Action.Actions.StatValue;
import com.al3x.housing2.Action.StatInstance;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

//public class StatActionData {
//    public static MoreStatData fromStatValue(StatValue statValue) {
//        return new MoreStatData(
////            statValue.getLiteralValue(),
////            statValue.getStatInstances(),
////            statValue.getValue() != null ? fromStatValue(statValue.getValue()) : null,
////            statValue.isExpression(),
////            statValue.getStatType()
//        );
//    }
//
//    @Getter
//    @Setter
//    public static class MoreStatData {
//        private String literal;
//        private List<StatInstance> statInstances;
//        private MoreStatData value;
//        private boolean isExpression;
//        private String statType;
//        private Boolean isGlobal;
//
//        public MoreStatData() {
//        }
//
//        public StatValue toStatValue() {
//            return new StatValue(
//                    isGlobal != null ? ((isGlobal) ? "global" : "player") : statType,
//                    isExpression,
//                    literal,
//                    value != null ? value.toStatValue() : null,
//                    statInstances
//            );
//        }
//    }
//}

File: src/main/java/com/al3x/housing2/Instances/HousingData/StatActionData.kt
package com.al3x.housing2.Instances.HousingData

import com.al3x.housing2.Action.Actions.StatValue
import com.al3x.housing2.Action.StatInstance
import com.al3x.housing2.Enums.StatOperation

class StatActionData {
    companion object {
        fun fromStatValue(statValue: StatValue): MoreStatData {
            return MoreStatData(
                statValue.literalValue,
                statValue.statInstances,
                statValue.value?.let { fromStatValue(it) },
                statValue.isExpression,
                statValue.isGlobal
            )
        }
    }
}

data class MoreStatData(
    val literal: String?,
    val statInstances: List<StatInstance>?,
    val value: MoreStatData?,
    val isExpression: Boolean,
    val isGlobal: Boolean = false
) {
    fun toStatValue(): StatValue {
        return StatValue(
            isGlobal,
            isExpression,
            literal,
            value?.toStatValue(),
            statInstances
        )
    }
}